plugins/modules/omevv_firmware.py

```diff
@@ -0,0 +1,1224 @@
+        exits with a failure message.
+
+        Parameters:
+            self (object): The instance of the class.
+
+        Returns:
+            None
+
+        Raises:
+            SystemExit: If the value of 'enter_maintenance_mode_timeout' is outside the valid range.
+        """
+        enter_maintenance_mode_timeout = self.module.params.get('enter_maintenance_mode_timeout')
+        if enter_maintenance_mode_timeout < 60 or enter_maintenance_mode_timeout > 1440:
+            self.module.exit_json(msg=MAINTENANCE_MODE_TIMEOUT_INVALID_MSG, failed=True)
+
+    def validate_params(self):
+        """
+        Validates the parameters of the class.
+
+        This function checks the existence of the 'host' and 'servicetag' parameters.
+        It also validates the 'job_wait' parameter and the 'date_time' parameter.
+        Additionally, it validates the 'enter_maintenance_mode_timeout' parameter.
+
+        Parameters:
+            self (object): The instance of the class.
+
+        Returns:
+            None
+        """
+
+        # Validate the 'host' and 'servicetag' parameter existence
+        self.host_servicetag_existence()
+
+        # Validate the job_wait parameter
+        if validate_job_wait(self.module):
+            self.module.exit_json(msg=TIMEOUT_NEGATIVE_OR_ZERO_MSG, failed=True)
+
+        # Validate the date_time parameter
+        if self.module.params.get('date_time'):
+            self.validate_date_time()
+
+        # Validate the enter_maintenance_mode_timeout parameter
+        if self.module.params.get('enter_maintenance_mode_timeout'):
+            self.enter_maintenance_mode_timeout()
+
+        param_result = True
+        return param_result
+
+
+class UpdateCluster(FirmwareUpdate):
+
+    def execute(self):
+        """
+        Executes the firmware update process.
+
+        This function validates the parameters, retrieves the target cluster and payload,
+        checks if the update job is allowed, and handles the firmware update process.
+
+        Returns:
+            None
+        """
+        cluster_name = None
+        cluster_group_id = None
+        host_service_tags = None
+        new_host_id = None
+        payload = None
+        self.validate_params()
+        vcenter_uuid = self.module.params.get('vcenter_uuid')
+        parameters = self.module.params
+        target = self.get_target(parameters['targets'])
+        cluster_name = target['cluster']
+
+        if target['cluster']:
+            cluster_group_id, payload, new_host_id = self.process_cluster_target(target)
+        else:
+            cluster_name, cluster_group_id, payload, new_host_id = self.process_non_cluster_target(parameters)
+
+        if not isinstance(new_host_id, list):
+            new_host_id = [new_host_id]
+            host_service_tags = [host_service_tags]
+
+        if not self.is_update_job_allowed(vcenter_uuid, cluster_group_id, cluster_name):
```

Comment:
This function will never be false, so this if statement will be true and it'll never return true, because in this function we have exit_json.

I would suggest remove return True in this function and replace line as below:
   self.is_update_job_allowed(vcenter_uuid, cluster_group_id, cluster_name)

Reply:
Incorporated the comment